pkg/controllers/vdb/passwordsecret_reconciler.go

```diff
@@ -63,89 +72,221 @@ func (a *PasswordSecretReconciler) Reconcile(ctx context.Context, _ *ctrl.Reques
 		return ctrl.Result{}, nil
 	}
 
-	// set status to the current using secret when db is initialized
-	if a.Vdb.Status.PasswordSecret == nil {
-		return ctrl.Result{}, a.updatePasswordSecretStatus(ctx)
+	sbName := a.PFacts.GetSandboxName()
+
+	// Ensure status initialized for main cluster or sandbox
+	if err := a.ensureStatusInitialized(ctx, sbName); err != nil {
+		return ctrl.Result{}, err
 	}
 
-	// No actions needed if status content is the same to spec
-	if a.statusMatchesSpec() {
+	// If we are reconciling a sandbox and it matches spec, no-op
+	// Also, if we are reconciling a sandbox and main cluster is not updated yet, no-op
+	if sbName != vapi.MainCluster {
+		if a.statusMatchesSpec(sbName) || !a.statusMatchesSpec(vapi.MainCluster) {
+			return ctrl.Result{}, nil
+		}
+	}
+
+	// If everything is up-to-date, no-op
+	outdatedSandboxes := a.getSandboxesNeedingUpdating()
```

Comment:
Do we need to calculate this when this reconciler is called by the sandbox controller?